test/integration/basic_test.go

```diff
@@ -0,0 +1,112 @@
+// +build integration
+
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/DATA-DOG/godog"
+	"github.com/DATA-DOG/godog/gherkin"
+
+	"github.com/minishift/minishift/test/integration/util"
+)
+
+type Minishift struct {
+	runner util.MinishiftRunner
+}
+
+func (m *Minishift) aPersona(personaName string) error {
+	// not implemented
+	return nil
+}
+
+func (m *Minishift) shouldBeRunning() error {
+	if !m.runner.CheckStatus("Running") {
+		return fmt.Errorf("Expected minishift to be running")
+	}
+
+	return nil
+}
+
+func (m *Minishift) shouldHaveAValidIPAddress() error {
+	ip := m.runner.RunCommand("ip", true)
+	ip = strings.TrimRight(ip, "\n")
+	if net.ParseIP(ip) == nil {
+		return fmt.Errorf("IP command returned an invalid address: %s", ip)
+	}
+
+	return nil
+}
+
+func (m *Minishift) executingCommand(command string) error {
+	m.runner.RunCommand(command, true)
+	return nil
+}
+
+func (m *Minishift) executingCommandWithReturn(command string, expected *gherkin.DocString) error {
+	expectedStr := expected.Content
+	cmdOutput := m.runner.RunCommand(command, true)
+	if !strings.Contains(cmdOutput, expectedStr) {
+		return fmt.Errorf("Output did not match. Expected: %s, Actual: %s", expectedStr, cmdOutput)
+	}
+
+	return nil
+}
+
+func FeatureContext(s *godog.Suite) {
+	runner := util.MinishiftRunner{
+		Args:       *args,
+		BinaryPath: *binaryPath}
+
+	m := &Minishift{runner: runner}
+
+	s.Step(`^a "([a-zA-Z\s]+)"$`, m.aPersona)
+	s.Step(`^should be running$`, m.shouldBeRunning)
```

Comment:
consistency... but the status is a possible improvement. 

the `process (.*)` seems in the described scenario less logical, as minishift itself is not the process running, but the hypervisor monitor process. Also, dockerd lives inside the VM?